pkg/sql/execinfrapb/processors_bulk_io.proto

```diff
@@ -198,6 +200,8 @@ message StreamIngestionDataSpec {
 
   // The processor will rekey the tenant's keyspace to a new tenant based on 'tenant_rekey'.
   optional TenantRekey tenant_rekey = 9 [(gogoproto.nullable) = false, (gogoproto.customname) = "TenantRekey"];
+
+  repeated jobs.jobspb.ResolvedSpan checkpoint = 10 [(gogoproto.nullable) = false];
```

Comment:
Do you think it might make sense to introduce a wrapper message (Checkpiont) and put resolved spans in there? Not sure if we'll wind up with more complex checkpointing -- I suspect yes.

Regardless of whether you agree with the above, I would like to see nice commend around this field given its relative importance.